wfe/wfe.go

```diff
@@ -1150,6 +1163,31 @@ func (wfe *WebFrontEndImpl) Registration(ctx context.Context, logEvent *requestE
 		return
 	}
 
+	// If key rollover has been enabled people may try to POST a different key
```

Comment:
I think this code specially handling the key field is unnecessary. The [spec](https://ietf-wg-acme.github.io/acme/#rfc.section.6.3) says:

> The server MUST ignore any values provided in the "key", "authorizations", and "certificates" fields in registration bodies sent by the client, as well as any other fields that it does not recognize. 

So we should just always clear the Key field on these inbound requests.

Alternately, we could treat mergeUpdate as being only for `POST /acme/reg/123` requests, and define a different RA method to rotate keys. This was the original purpose of mergeUpdate, to whitelist only the specific fields that are updateable by user POSTs. That said, I see why you made the choice to reuse meregeUpdate (/UpdateRegistration), since it avoids having to create another RPC. @cpu, what do you think?